src/ServiceControl.Persistence.RavenDb/UnitOfWork/RavenDbRecoverabilityIngestionUnitOfWork.cs

```diff
@@ -57,21 +57,41 @@ public Task RecordSuccessfulRetry(string retriedMessageUniqueId)
             var serializedGroups = RavenJToken.FromObject(groups);
             var serializedAttempt = RavenJToken.FromObject(processingAttempt, Serializer);
 
+            //HINT: RavenDB5 is using Lodash v4.13.1 to provide javascript utility functions
+            //      https://ravendb.net/docs/article-page/3.5/csharp/client-api/commands/patches/how-to-use-javascript-to-patch-your-documents#methods-objects-and-variables
             return new ScriptedPatchCommandData
             {
                 Key = documentId,
                 Patch = new ScriptedPatchRequest
                 {
                     Script = $@"this.{nameof(FailedMessage.Status)} = status;
                                 this.{nameof(FailedMessage.FailureGroups)} = failureGroups;
+                                
+                                var newAttempts = this.{nameof(FailedMessage.ProcessingAttempts)};
+
+                                //De-duplicate attempts by AttemptedAt value
 
                                 var duplicateIndex = _.findIndex(this.{nameof(FailedMessage.ProcessingAttempts)}, function(a){{
                                     return a.{nameof(FailedMessage.ProcessingAttempt.AttemptedAt)} === attempt.{nameof(FailedMessage.ProcessingAttempt.AttemptedAt)};
                                 }});
 
                                 if(duplicateIndex === -1){{
-                                    this.{nameof(FailedMessage.ProcessingAttempts)} = _.union(this.{nameof(FailedMessage.ProcessingAttempts)}, [attempt]);
-                                }}",
+                                    newAttempts = _.union(newAttempts, [attempt]);
+                                }}
+
+                                //Trim to the latest MaxProcessingAttempts 
+                                
+                                newAttempts = _.sortBy(newAttempts, function(a) {{
+                                    return a.{nameof(FailedMessage.ProcessingAttempt.AttemptedAt)};
+                                }});
+                                
+                                if(newAttempts.length > {MaxProcessingAttempts})
```

Comment:
We need to change this to ensure that the already existing collection with more than maxprocessingattempts are truncated correctly.